internal/unmarshal.go

```diff
@@ -38,135 +39,92 @@ func convertNumber[T Numbers](anyNumber any) (T, error) {
 	}
 }
 
-func convertSliceyNumbers[T Numbers](anySlice []any) ([]T, error) {
-	typedSlice := make([]T, len(anySlice))
-	for i, v := range anySlice {
-		convRes, convErr := convertNumber[T](v)
-		if convErr != nil {
-			return nil, fmt.Errorf("error converting number-slice element: %w", convErr)
-		}
-		typedSlice[i] = convRes
-	}
-	return typedSlice, nil
-}
-
-func convertSliceyNonNumbers[T any](anySlice []any) ([]T, error) {
-	typedSlice := make([]T, len(anySlice))
-	for i, v := range anySlice {
-		castRes, ok := v.(T)
-		if !ok {
-			var t T
-			return []T{}, fmt.Errorf("cannot cast the slice element '%s' of type '%s' to the specified type '%s'", v, reflect.TypeOf(v), reflect.TypeOf(t))
+func isTypeDataclass(typ reflect.Type) bool {
```

Comment:
I'm a bit unfamiliar with this part of the codebase, but why do we care specifically if something is a dataclass vs say another type of struct?

Reply:
So that we can distinguish feature classes from dataclasses, and handle unmarshalling them differently.